### PR TITLE
Hw 33

### PR DESCRIPTION
**Pull request by Oleh888**

### CODE REVIEW

File: src/main/java/com/dev/cinema/controller/CinemaHallController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+@RestController
+@RequestMapping(value = "/cinema-halls")
+public class CinemaHallController {
+    private final CinemaHallMapper cinemaHallMapper;
+
+    public CinemaHallController(CinemaHallMapper cinemaHallMapper) {
+        this.cinemaHallMapper = cinemaHallMapper;
+    }
+
+    @PostMapping(value = "/add")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/CinemaHallController.java:23]:
```suggestion
    @PostMapping
```
image

File: src/main/java/com/dev/cinema/controller/CinemaHallController.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    public CinemaHall addCinemaHall(@RequestBody CinemaHallRequestDto cinemaHallRequestDto) {
+        return cinemaHallMapper.getCinemaHallFromRequestDto(cinemaHallRequestDto);
+    }
+
+    @GetMapping(value = "/all")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/CinemaHallController.java:28]:
```suggestion
    @GetMapping
```

File: src/main/java/com/dev/cinema/controller/MovieController.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+@RestController
+@RequestMapping(value = "/movies")
+public class MovieController {
+    private final MovieMapper movieMapper;
+
+    public MovieController(MovieMapper movieMapper) {
+        this.movieMapper = movieMapper;
+    }
+
+    @PostMapping(value = "/add")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/MovieController.java:23]: same

File: src/main/java/com/dev/cinema/controller/MovieController.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    public Movie addMovie(@RequestBody MovieRequestDto movieRequestDto) {
+        return movieMapper.getMovieFromRequestDto(movieRequestDto);
+    }
+
+    @GetMapping(value = "/all")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/MovieController.java:28]: same

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+@RestController
+@RequestMapping(value = "/movie-sessions")
+public class MovieSessionController {
+    private final MovieSessionMapper movieSessionMapper;
+
+    public MovieSessionController(MovieSessionMapper movieSessionMapper) {
+        this.movieSessionMapper = movieSessionMapper;
+    }
+
+    @PostMapping(value = "/add")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/MovieSessionController.java:24]:
```suggestion
    @PostMapping
```

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    public OrderController(OrderMapper orderMapper) {
+        this.orderMapper = orderMapper;
+    }
+
+    @PostMapping(value = "/complete")
+    public Order completeOrder(@RequestParam Long userId) {
+        return orderMapper.getOrderFromUserId(userId);
+    }
+
+    @GetMapping(value = "/all")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/OrderController.java:27]:
```suggestion
    @GetMapping
```

File: src/main/java/com/dev/cinema/dao/MovieDao.java [hunk 7]
```diff
@@ -7,4 +7,6 @@
     Movie add(Movie movie);
 
     List<Movie> getAll();
+
+    Movie getByMovieId(Long id);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/MovieDao.java:11]:
```suggestion
    Movie getById(Long id);
```

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 8]
```diff
@@ -52,4 +55,19 @@ public CinemaHall add(CinemaHall cinemaHall) {
             throw new DataProcessingException("Can't get all cinema hall", e);
         }
     }
+
+    @Override
+    public CinemaHall getByCinemaHallId(Long id) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java:60]:
```suggestion
    public CinemaHall getById(Long id) {
```
you are making these methods in corresponding daos, so there is no reason to specify which id in method's name(same everywhere)

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 9]
```diff
@@ -52,4 +55,19 @@ public CinemaHall add(CinemaHall cinemaHall) {
...
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<CinemaHall> criteriaQuery =
+                    criteriaBuilder.createQuery(CinemaHall.class);
+            Root<CinemaHall> root = criteriaQuery.from(CinemaHall.class);
+            Predicate predicateForId = criteriaBuilder.equal(root.get("id"), id);
+            criteriaQuery.where(predicateForId);
+            return session.createQuery(criteriaQuery).uniqueResult();
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java:68]:
```suggestion
          return session.get(CinemaHall.class, id);
```
try this one(in other daos too)

File: src/main/java/com/dev/cinema/controller/MovieController.java (hunk 3 above)
Comment (Oleh888) [reply src/main/java/com/dev/cinema/controller/MovieController.java:23]: done

File: src/main/java/com/dev/cinema/controller/MovieController.java (hunk 4 above)
Comment (Oleh888) [reply src/main/java/com/dev/cinema/controller/MovieController.java:28]: done

**Pull request merged**